index.d.ts

```diff
@@ -0,0 +1,10 @@
+declare module 'objection-unique' {
+  import { Model } from 'objection';
+
+  type Constructor<T = {}> = new (...args: any[]) => T;
+
+  export default function (options?: {
+    identifiers?: string[];
+    fields?: string[];
+  }): <T extends typeof Model>(model: T) => T & Constructor<T>
```

Comment:
Taking in account the compatibility with other `objection` plugins, the return type should be just the provided model. 

Example: https://github.com/oscaroox/objection-timestamps/blob/master/src/index.ts#L25.

Reply:
I followed objection timestamp implementation to return the model itself.
Thanks for all the feedback!